pkg/netpol/connlist/conns_formatter.go

```diff
@@ -67,3 +69,56 @@ func (j jsonFormatter) writeOutput(conns []Peer2PeerConnection) (string, error)
 	}
 	return string(jsonConns), nil
 }
+
+// dotFormatter: implements the connsFormatter interface for dot output format
+type dotFormatter struct {
+}
+
+// formats an edge line from a singleConnFields struct , to be used for dot graph
+func (d dotFormatter) getEdgeLine(c singleConnFields) string {
+	return fmt.Sprintf("\t%q -> %q [label=%q color=\"gold2\" fontcolor=\"darkgreen\"]\n", c.Src, c.Dst, c.ConnString)
+}
+
+// formats a peer line for dot graph
+func (d dotFormatter) getPeerLine(peer eval.Peer) string {
+	var peerColor string
+	if peer.IsPeerIPType() {
+		peerColor = "red2"
+	} else {
+		peerColor = "blue"
+	}
+	peerName := peer.String()
+	return fmt.Sprintf("\t%q [label=%q color=%q fontcolor=%q]\n", peerName, peerName, peerColor, peerColor)
+}
+
+// sorts the final dot graph lines
+func (d dotFormatter) sortGraphLines(graphLines string) string {
+	var graphLinesSorted sort.StringSlice = strings.Split(graphLines, "\n")
+	graphLinesSorted.Sort()
+	return strings.Join(graphLinesSorted, "\n")
+}
+
+// returns a dot string form of connections from list of Peer2PeerConnection objects
+func (d dotFormatter) writeOutput(conns []Peer2PeerConnection) (string, error) {
+	edgeLines := make([]string, len(conns))
+	peerLines := make(map[string]string, 0)
+	for index := range conns {
+		connLine := formSingleConn(conns[index])
+		edgeLines[index] = d.getEdgeLine(connLine)
+		if _, ok := peerLines[connLine.Src]; !ok {
+			peerLines[connLine.Src] = d.getPeerLine(conns[index].Src())
+		}
+		if _, ok := peerLines[connLine.Dst]; !ok {
+			peerLines[connLine.Dst] = d.getPeerLine(conns[index].Dst())
+		}
+	}
+	var graphLines string
+	for _, peerLine := range peerLines {
+		graphLines += peerLine
+	}
+	for _, edgeLine := range edgeLines {
+		graphLines += edgeLine
+	}
+	res := "digraph {" + d.sortGraphLines(graphLines) + "\n}\n"
+	return res, nil
```

Comment:
Should sort separately edge lines and peer lines , so that in the output peer lines are placed before edge lines.
Also, avoid having the current flow of strings concatenation, then splitting, then concatenating again.